lib/rspec/matchers.rb

```diff
@@ -1005,5 +1005,31 @@ def self.is_a_matcher?(obj)
     def self.is_a_describable_matcher?(obj)
       is_a_matcher?(obj) && obj.respond_to?(:description)
     end
+
+    if RSpec::Support::Ruby.mri? && RUBY_VERSION[0, 3] == '1.9'
+      # @api private
+      # Note that `included` doesn't work for this because it is triggered
+      # _after_ `RSpec::Matchers` is an ancestor of the inclusion host, rather
+      # than _before_, like `append_features`. It's important we check this before
+      # in order to find the cases where it was already previously included.
+      def self.append_features(mod)
+        subclasses = ObjectSpace.each_object(Class).select { |c| c < mod && c < self }
+        return super unless subclasses.any?
+
+        subclasses.reject! { |s| subclasses.any? { |s2| s < s2 } } # Filter to the root ancestor.
+        subclasses = subclasses.map { |s| "`#{s}`" }.join(", ")
+
+        RSpec.warning "`#{self}` has been included in a superclass (`#{mod}`) " \
+                      "after previously being included in subclasses (#{subclasses}), " \
```

Comment:
That was a bug, actually, related to the fact that I didn't handle re-inclusion properly.